src/install/dracut-install.c

```diff
@@ -618,7 +619,15 @@ static int resolve_deps(const char *src)
         }
 
         fclose(fptr);
-        wait(&err);
+        while (waitpid(ldd_pid, &err, 0) == -1) {
+            if (errno != EINTR) {
+                if (err != 0) {
```

Comment:
`waitpid()` returns the terminated pid (> 0) when sucessfull, but its exit status still can be abnormal: non-zero exit code, killed by signal... hence `err != 0` check and error message print need to be outside the loop[*]. If `waitpid() < 0` and `errno != EINTR` things are broken and we must assume everything failed: `err` in this case *cannot* be trusted (I think most implementations do not change it, not sure). Set it to anything non-zero.

`write(2, "command not found", ...)` is not needed. Also, to print meaningful exit codes/signals in the error message, it would need WIFEXITED, WEXITSTATUS, WIFSIGNALED, WTERMSIG, macros. Overkill in my opinion.

[*] `cp()` function (`normal_copy` label) is buggy in this regard too, because those `log_error()` will not run in case of normal errors, when `waitpid()` succeeds (missing `cp` binary, etc).

Reply:
I was clearly in no state to be writing this last night, because this is an obvious error; I've fixed this and cp.